src/sdk-client.ts

```diff
@@ -0,0 +1,327 @@
+        const api = new LifecycleStatesApi(this.config)
+
+        const response = await api.listLifecycleStates({ identityProfileId })
+
+        return response.data
+    }
+
+    async listWorkgroupMembers(workgroupId: string): Promise<ListWorkgroupMembers200ResponseInnerV2[]> {
+        const api = new GovernanceGroupsV2Api(this.config)
+        const response = await api.listWorkgroupMembers({ workgroupId })
+
+        return response.data
+    }
+
+    async listIdentities(): Promise<IdentityBeta[]> {
+        const api = new IdentitiesBetaApi(this.config)
+
+        const response = await Paginator.paginate(api, api.listIdentities)
+
+        return response.data
+    }
+
+    async listPrivilegedIdentities(): Promise<IdentityDocument[]> {
+        const api = new SearchApi(this.config)
+        const search: Search = {
+            indices: ['identities'],
+            query: {
+                query: '@access(source.name.exact:IdentityNow)',
+            },
+            sort: ['id'],
+            includeNested: true,
+        }
+
+        const response = await Paginator.paginateSearchApi(api, search)
+
+        return response.data
+    }
+
+    async listAccountsByIdentity(id: string): Promise<Account[]> {
+        const api = new AccountsApi(this.config)
+
+        const filters = `identityId eq "${id}"`
+        const listAccountsByIdentity = (
+            requestParameters?: AccountsApiListAccountsRequest,
+            axiosOptions?: AxiosRequestConfig
+        ): Promise<import('axios').AxiosResponse<Account[], any>> => {
+            return api.listAccounts({ filters })
+        }
+        const response = await Paginator.paginate(api, listAccountsByIdentity)
+
+        return response.data
+    }
+
+    async getAccountDetails(id: string): Promise<IdentityBeta> {
+        const api = new IdentitiesBetaApi(this.config)
+
+        const requestParameters: IdentitiesBetaApiGetIdentityRequest = {
+            id,
+        }
+        const response = await api.getIdentity(requestParameters)
+
+        return response.data
+    }
+
+    async addWorkgroup(id: string, workgroupId: string): Promise<void> {
+        const api = new GovernanceGroupsV2Api(this.config)
+
+        const requestParameters: GovernanceGroupsV2ApiModifyWorkgroupMembersRequest = {
+            workgroupId,
+            modifyWorkgroupMembersRequestV2: { add: [id] },
+        }
+        const response = await api.modifyWorkgroupMembers(requestParameters)
+
+        await sleep(2000)
+        return response.data
+    }
+
+    async removeWorkgroup(id: string, workgroupId: string): Promise<void> {
+        const api = new GovernanceGroupsV2Api(this.config)
+
+        const requestParameters: GovernanceGroupsV2ApiModifyWorkgroupMembersRequest = {
+            workgroupId,
+            modifyWorkgroupMembersRequestV2: { remove: [id] },
+        }
+        const response = await api.modifyWorkgroupMembers(requestParameters)
+
+        await sleep(2000)
```

Comment:
Why the need to sleep here?

Reply:
From experience, some operations take a few seconds to execute and show the new state after change. If I don't include these short sleeps at some points, when reading back the changes I introduce to respond to IDN, the results are incomplete. At some point I will change the way I compose my object results so I don't have to read back from IDN but, for the moment, I read everything back so this is needed.